zio-http/src/main/scala/zio/http/Driver.scala

```diff
@@ -2,31 +2,10 @@ package zio.http
 
 import zio._
 
-import java.util.concurrent.atomic.AtomicReference
-
 trait Driver {
   def start(): RIO[Scope, Int]
-}
-
-object Driver {
-
-  private[zio] final case class Context private (
-    httpAppRef: AtomicReference[HttpApp[Any, Throwable]],
-    errorCallbackRef: AtomicReference[Option[Server.ErrorCallback]],
-  ) {
-
-    def errorCallback: Option[Server.ErrorCallback] = errorCallbackRef.get
-
-    def onApp[A](f: HttpApp[Any, Throwable] => A): A = f(httpAppRef.get())
-
-    def onAppRef[A](f: AtomicReference[HttpApp[Any, Throwable]] => A): A = f(httpAppRef)
-
-    def onErrorCallbackRef[A](f: AtomicReference[Option[Server.ErrorCallback]] => A): A = f(errorCallbackRef)
-
-  }
 
-  object Context {
+  def setErrorCallback(newCallback: Option[Server.ErrorCallback]): UIO[Unit]
 
-    private[zio] def empty = Context(new AtomicReference(Http.empty), new AtomicReference(Option.empty))
-  }
+  def setApp(newApp: HttpApp[Any, Throwable]): UIO[Unit]
```

Comment:
`setApp` implies that there is a single app that should be set one time.

Instead, the feature needs to be: `addApp`, which adds the app together with any existing app. These are the semantics required by `Server#install`.

Reply:
Done.